Mage/src/main/java/mage/game/GameState.java

```diff
@@ -819,19 +818,31 @@ public boolean hasSimultaneousEvents() {
     public void addSimultaneousDamage(DamagedEvent damagedEvent, Game game) {
         // combine damages per type (player or permanent)
         boolean flag = false;
+        Set<UUID> flagPerPlayer = new HashSet<>();
```

Comment:
can you add comments explaining how these flags are used?

Reply:
And actually I do not need the full Set, just a boolean that the batched played for damagedEvent has been found or not. I'll cleanup that part.

Reply:
Comments look good, can you remind me why just a boolean instead of a set?

Reply:
Oh I did comment on your other message, we are looking for a specific batch for the player. This method is adding 1 damage event to simultaneous event, and adding/modifying the relevant batches.
Storing the Set was overkill. But there is a missing condition here, I'll fix it in a few minutes.